pandas/core/dtypes/dtypes.py

```diff
@@ -71,6 +72,18 @@ def __eq__(self, other):
     def __ne__(self, other):
         return not self.__eq__(other)
 
+    def __getstate__(self):
+        # pickle support; we don't want to pickle the cache
+        return {k: getattr(self, k, None) for k in self._metadata}
+
+    def __setstate__(self, state):
+        self.__dict__.update(state)
```

Comment:
FWIW this is the [default behavior](https://docs.python.org/3/library/pickle.html#object.__setstate__) if there is no `__setstate__` method